CMakePresets.json

```diff
@@ -0,0 +1,60 @@
+{
+    "version": 5,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 24,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "base-dev",
+            "binaryDir": "${sourceDir}/build/${presetName}",
+            "hidden": true
+        },
+        {
+            "name": "msvc-17",
+            "inherits": "base-dev",
+            "displayName": "Visual Studio 2022",
+            "generator": "Visual Studio 17 2022",
+            "description": "Configure using Visual Studio 17 2022 generator",
+            "condition": {
+                "type": "equals",
+                "lhs": "${hostSystemName}",
+                "rhs": "Windows"
+            },
+            "cacheVariables": {
+                "CMAKE_MSVC_RUNTIME_LIBRARY": "MultiThreadedDLL"
```

Comment:
And again, assume we can't submit this upstream to Corrosion until rustc / cc have fixes ? Eg is there a reason why Corrosion can't set `CMAKE_MSVC_RUNTIME_LIBRARY = MultiThreadedDLL` for all projects on Windows ? Or does this break other projects ? :thinking: 

As for us that would then help projects that are using CXX-Qt as a library as they don't need to remember to set this in their CMake as well.

Reply:
From the [documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#:~:text=the%20MSVC_RUNTIME_LIBRARY%20property%20on%20all%20targets%20as%20they%20are%20created) changing CMAKE_MSVC_RUNTIME_LIBRARY  from corrosion after creating any target would not work for target created before changing CMAKE_MSVC_RUNTIME_LIBRARY .
I haven't tried that it's just my understanding from the doc.
Once again, writing to a global variable in not a good idea